core/storage/trie_pruner/impl/dummy_pruner.hpp

```diff
@@ -43,6 +43,8 @@ namespace kagome::storage::trie_pruner {
     std::optional<uint32_t> getPruningDepth() const override {
       return std::nullopt;
     }
+
+    void reload(const blockchain::BlockTree &) override {}
```

Comment:
I don't think reload is a descriptive name. Better just make restoreStateAt public. Also, I think that it should be protected by a mutex now that it can be used at a random moment.